src/server/pkg/storage/fileset/writer.go

```diff
@@ -264,17 +207,28 @@ func (w *Writer) Close() error {
 	if w.noUpload {
 		return nil
 	}
-	// Close the index writer.
-	idx, err := w.iw.Close()
+	// Close the index writers.
+	additiveIdx, err := w.additive.Close()
+	if err != nil {
+		return err
+	}
+	deletiveIdx, err := w.deletive.Close()
 	if err != nil {
 		return err
 	}
 	var pointsTo []string
-	for _, cid := range index.PointsTo(idx) {
+	for _, cid := range index.PointsTo(additiveIdx) {
+		pointsTo = append(pointsTo, chunk.ObjectID(cid))
+	}
+	for _, cid := range index.PointsTo(deletiveIdx) {
 		pointsTo = append(pointsTo, chunk.ObjectID(cid))
 	}
 	if err := w.tracker.CreateObject(w.ctx, filesetObjectID(w.path), pointsTo, w.ttl); err != nil {
 		return err
```

Comment:
What do we want to do if the fileset already exists?  `tracker.CreateObject` will return an err, but w.store.Set should be fine.

Reply:
Wouldn't it be an error to try and create a file set that already exists?

Reply:
The issue is if you successfully create the object, but you don't actually finish adding it to the fileset table.  If the paths are write once, the correct thing to do is ignore `tracker.ErrObject` exists and recreate the object.  If the paths are not write once then we could be in an inconsistent state after an error.

Reply:
Yeah, I think we need to have these in a single transaction. I added a TODO for this.